src/dft/backends/rocfft/backward.cpp

```diff
@@ -0,0 +1,264 @@
+/*******************************************************************************
+* Copyright Codeplay Software Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include <rocfft.h>
+#include <hip/hip_runtime_api.h>
+
+#include "execute_helper.hpp"
+#include "oneapi/mkl/dft/backward.hpp"
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "oneapi/mkl/dft/types.hpp"
+#include "oneapi/mkl/exceptions.hpp"
+#include "rocfft_handle.hpp"
+
+namespace oneapi::mkl::dft::rocfft {
+namespace detail {
+template <dft::precision prec, dft::domain dom>
+rocfft_plan get_bwd_plan(dft::detail::commit_impl<prec, dom> *commit) {
+    return static_cast<rocfft_handle *>(commit->get_handle())[1].plan.value();
+}
+
+template <dft::precision prec, dft::domain dom>
+rocfft_execution_info get_bwd_info(dft::detail::commit_impl<prec, dom> *commit) {
+    return static_cast<rocfft_handle *>(commit->get_handle())[1].info.value();
+}
+} // namespace detail
+// BUFFER version
+
+//In-place transform
+template <typename descriptor_type, typename data_type>
+ONEMKL_EXPORT void compute_backward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout) {
+    detail::expect_config<dft::config_param::PLACEMENT, dft::config_value::INPLACE>(
+        desc, "Unexpected value for placement");
+    auto commit = detail::checked_get_commit(desc);
+    auto queue = commit->get_queue();
+    auto plan = detail::get_bwd_plan(commit);
+    auto info = detail::get_bwd_info(commit);
+
+    queue.submit([&](sycl::handler &cgh) {
+        auto inout_acc = inout.template get_access<sycl::access::mode::read_write>(cgh);
+
+        cgh.host_task([=](sycl::interop_handle ih) {
+            const std::string func_name = "compute_backward(desc, inout)";
+            auto stream = detail::setup_stream(func_name, ih, info);
```

Comment:
is the stream used to syncronize many computes ?

Reply:
The `interop_handle` will give us a stream (not guaranteed to be the same one each time), and I use that to synchronize so that when the `host_task` is finished, the computation is also complete.
I believe the backend will create a large number (>100) stream, and pass then out in a round robin fashion (but this is implementation defined).